crates/goose/src/providers/azure.rs

```diff
@@ -63,16 +63,26 @@ impl AzureProvider {
     }
 
     async fn post(&self, payload: Value) -> Result<Value, ProviderError> {
-        let url = format!(
-            "{}/openai/deployments/{}/chat/completions?api-version={}",
-            self.endpoint.trim_end_matches('/'),
-            self.deployment_name,
-            AZURE_API_VERSION
-        );
+        let base_url = url::Url::parse(&self.endpoint.trim_end_matches('/'))
+            .map_err(|e| ProviderError::RequestFailed(format!("Invalid base URL: {e}")))?;
```

Comment:
build fails - i think we can remove `.trim_end_matches('/')` since we use url join